src/navigation/search.py

```diff
@@ -101,9 +98,11 @@ def evaluate(self, ud):
         self.context.rover.send_drive_command(cmd_vel)
 
         # if we see the fiduicial or gate, go to either fiducial or gate state
-        if self.context.env.current_gate() is not None:
+        if self.context.env.current_gate() is not None and self.context.course.look_for_gate():
             return SearchStateTransitions.found_gate.name  # type: ignore
-        elif self.context.env.current_fid_pos() is not None:
-            return SearchStateTransitions.found_fiducial.name  # type: ignore
-
+        elif self.context.env.current_fid_pos() is not None or self.context.env.other_gate_fid_pos() is not None:
+            if self.context.course.look_for_post():
+                return SearchStateTransitions.found_fiducial_post.name  # type: ignore
+            else:
```

Comment:
if we're looking for a post and we see a tag that has id 1 greater than the current tag we're looking for (I think this is the value of the next waypoint in our course), wouldn't this falsely recognize that as the other_gate_fid_pos and then move into found_fiducial_post state even though we haven't found the one we want?

Reply:
yea this is a good catch. if you change the other_gate_fid function to return None if we aren't in gate mode that should fix it

Reply:
yeah I'm gonna change other_gate_fid as well as the get_gate function to check if we are actually in gate mode

Reply:
yeah I'm gonna change other_gate_fid as well as the get_gate function to check if we are actually in gate mode